### PR TITLE
Allow multiple sinks for reduce type plugins

### PR DESCRIPTION
**Pull request by CuriousVini**
Issue: https://issues.cask.co/browse/HYDRA-48
Build: http://builds.cask.co/browse/CDAP-DUT4539-4

### CODE REVIEW

File: cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/mapreduce/TransformRunner.java [hunk 1]
```diff
@@ -129,7 +129,8 @@ public TransformRunner(MapReduceTaskContext<Object, Object> context,
     Set<StageInfo> reducers = pipelinePhase.getStagesOfType(BatchAggregator.PLUGIN_TYPE, BatchJoiner.PLUGIN_TYPE);
     if (!reducers.isEmpty()) {
       String reducerName = reducers.iterator().next().getName();
-      if (pipelinePhase.getSinks().contains(reducerName)) {
+      JobContext hadoopContext = context.getHadoopContext();
+      if (pipelinePhase.getSinks().contains(reducerName) || hadoopContext instanceof Mapper.Context) {
```
Comment (albertshau) [inline cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/mapreduce/TransformRunner.java]: I don't understand this logic. If this is a mapper with multiple sinks, we need a MultiOutputWriter right?  If the 'sink' is a reduce type plugin, isn't that enough to know that it should be single output?

Comment (albertshau) [inline cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/mapreduce/TransformRunner.java]: ok I read through the code again, this is not where the fix should be and will in fact introduce bugs. Doesn't this cause unit tests to fail?

Instead, you need to change the TransformRunner to call this method after it does the phase subsets.

**Comment by albertshau** [comment]
lgtm

**Comment by CuriousVini** [comment]
Build is green. Squashing and merging.

**Pull request merged**